### PR TITLE
DM info in linelist

### PR DESCRIPTION
**Pull request by harimohanraj89**
**Story card:** https://www.pivotaltracker.com/story/show/173298755

## Because

We want to include blood sugars in the line list

## This addresses

This PR does the following things to the patient line list download:

* Makes the "Latest BP Date" the first column in the "Latest BP" section
* Add three new columns to the patient line list: 
  * "Latest Blood Sugar Date"
  * "Latest Blood Sugar Value" (eg. `150 mg/dL`)
  * "Latest Blood Sugar Type" (eg. `Random`)
* Populates ~blood sugar data into the new columns if and only if the patient had a blood sugar reading on the same day as their latest BP~ patient's latest blood sugar with date

## Additional notes

* This PR also tackles a few N+1 queries that were remaining in the line list. The line list query should now be free of N+1's.
* In order to combat the memory usage spike from avoiding N+1 queries, we've reduced the patient batch size `1000 -> 20`

### CODE REVIEW

File: app/exporters/patients_exporter.rb [hunk 1]
```diff
@@ -16,10 +30,10 @@ def self.csv(patients)
           :phone_numbers,
           :address,
           :medical_history,
-          :prescription_drugs,
+          :current_prescription_drugs,
           :latest_bp_passports,
           { latest_scheduled_appointments: :facility },
-          { latest_blood_pressures: :facility },
+          { latest_blood_pressures: [:facility, {encounter: :blood_sugars}] },
```
Comment (prabhanshuguptagit) [inline app/exporters/patients_exporter.rb:36]: > Populates blood sugar data into the new columns if and only if the patient had a blood sugar reading on the same day as their latest BP

So if a patient has only a Blood sugar recorded, we will not add a row in the line list? Could you point me to some context around this, a bit out of the loop.

Comment (prabhanshuguptagit) [inline app/exporters/patients_exporter.rb:36]: > Populates blood sugar data into the new columns if and only if the patient had a blood sugar reading on the same day as their latest BP

Is there a reason to not add patients who have just blood sugars? Sorry a bit out of the loop on the discussion around this.

File: spec/factories/blood_sugars.rb [hunk 2]
```diff
@@ -21,12 +21,24 @@
       end
     end
 
-    trait :with_hba1c do
-      blood_sugar_type { BloodSugar::blood_sugar_types.keys.sample }
```
[the remark this answers is not in the document]
Comment (harimohanraj89) [reply spec/factories/blood_sugars.rb:25]: Oops! I mistakenly grepped for this trait only in `spec/`. I see it now, I'll leave this one untouched, and just add the other traits that _fix_ the type.

File: app/exporters/patients_exporter.rb [hunk 3]
```diff
@@ -5,6 +5,20 @@ module PatientsExporter
 
   BATCH_SIZE = 1000
 
+  BLOOD_SUGAR_UNITS = {
```
[the remark this answers is not in the document]
Comment (harimohanraj89) [reply app/exporters/patients_exporter.rb:8]: I thought about it. We don't render this stuff anywhere else on the server. For now, I thought we'll keep it closer to the presentation layer, and then pull it closer to the model as more use cases for it come up.

File: app/exporters/patients_exporter.rb (hunk 1 above)
Comment (harimohanraj89) [reply app/exporters/patients_exporter.rb:36]: Totally. We've established that _most_ of our patients will definitely have a BP along with their blood sugar reading: https://simpledotorg.slack.com/archives/CFJMFSM8F/p1592304255188700

So this PR is just meant to _enrich_ the hypertension line list.

File: app/exporters/patients_exporter.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (prabhanshuguptagit) [reply app/exporters/patients_exporter.rb:8]: ✌️

File: spec/exporters/patients_exporter_spec.rb [hunk 4]
```diff
@@ -136,5 +162,13 @@
       expect(subject.csv_headers).not_to include("Patient #{Address.human_attribute_name :zone}")
       expect(subject.csv_fields(patient)).not_to include(patient.address.zone)
     end
+
+    it "does not include blood sugars from older appointments" do
```
Comment (prabhanshuguptagit) [inline spec/exporters/patients_exporter_spec.rb:166]: nitpick: older visits?

Comment (harimohanraj89) [reply spec/exporters/patients_exporter_spec.rb:166]: Good catch 👍

**Pull request merged**